store/src/main/java/com/book/store/service/impl/UserDetailsServiceImpl.java

```diff
@@ -0,0 +1,28 @@
+package com.book.store.service.impl;
+
+import com.book.store.model.User;
+import com.book.store.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserDetailsServiceImpl implements UserDetailsService {
+
+    private final UserService userService;
```

Comment:
it's odd that service uses another service to access DB
```suggestion
    private final UserRepository userRepository;
```